server/src/main/scala/scaladex/server/service/Scheduler.scala

```diff
@@ -51,8 +55,28 @@ class Scheduler(val name: String, runnable: Runnable)(implicit ec: ExecutionCont
     status match {
       case s: SchedulerStatus.Started =>
         cancellable.map(_.cancel())
-        _status = SchedulerStatus.Stopped(Instant.now)
+        _status = SchedulerStatus.Stopped(name, Instant.now)
         cancellable = None
       case _ => ()
     }
+
+  private def runnable: Runnable = new Runnable {
+    override def run(): Unit = {
+      logger.info(s"Scheduler ${name}: Starting at ${Instant.now}")
+      val future: Future[Unit] =
+        try for {
+          _ <- job()
+        } yield logger.info(s"Scheduler $name: Finished running the scheduler ${Instant.now}")
+        catch {
+          case NonFatal(e) =>
+            Future.successful(
+              logger.info(
+                s"Scheduler ${name}: Failed because ${e.getMessage}"
+              )
+            )
+            Future.successful(logger.info(e.getStackTrace.mkString("\n")))
+        }
+      Await.result(future, Duration.Inf)
```

Comment:
The `try {} catch` here will never catch anything because it happens inside a future.
Also the logs inside the catch should probably be warnings and it is not really useful to get the `Instant.now` in the logs since all logs are already dated.


```suggestion
      logger.info(s"Scheduler ${name}: Starting")
       try {
         Await.result(job(), Duration.Inf)
         logger.info("Scheduler $name: Finished")
       } catch {
         case NonFatal(e) =>
            logger.warning(s"Scheduler ${name}: Failed because ${e.getMessage}")
            logger.warning(e.getStackTrace.mkString("\n")))
       }
```